lfs/transfer_queue.go

```diff
@@ -17,45 +18,52 @@ const (
 )
 
 type Transferable interface {
-	Check() (*api.ObjectResource, error)
-	Transfer(progress.CopyCallback) error
-	Object() *api.ObjectResource
 	Oid() string
 	Size() int64
 	Name() string
+	Path() string
+	Object() *api.ObjectResource
 	SetObject(*api.ObjectResource)
+	// Legacy API check - TODO remove this and only support batch
+	LegacyCheck() (*api.ObjectResource, error)
 }
 
-// TransferQueue provides a queue that will allow concurrent transfers.
+// TransferQueue organises the wider process of uploading and downloading,
+// including calling the API, passing the actual transfer request to transfer
+// adapters, and dealing with progress, errors and retries
 type TransferQueue struct {
```

Comment:
This type is getting to hold a lot of state. Are there perhaps smaller sub-types that we could pull out of TransferQueue, and then it could simply orchestrate those types?